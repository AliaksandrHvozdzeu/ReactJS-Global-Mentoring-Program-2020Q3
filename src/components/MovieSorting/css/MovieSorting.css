.sorting-title {
  color: #d6cbcb;
  font-weight: 100;
  top: 20px;
  right: 0;
  position: absolute;
}

.sorting-select {
  background-color: var(--grey-darker);
  color: var(--white);
  font-size: 1.2rem;
  outline: none;
  margin-left: 2rem;
  text-transform: uppercase;
}

.sorting-select:hover {
  cursor: pointer;
}

.sorting-select::after {
  content: '';
  border-top: 5px solid var(--red);
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-bottom: 5px solid transparent;
  position: absolute;
  top:50%;
  left: 50%;
  z-index: 2;
}
