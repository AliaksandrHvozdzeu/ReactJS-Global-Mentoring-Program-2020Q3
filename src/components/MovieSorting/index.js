import MovieSorting from './MovieSorting';

export default MovieSorting;
