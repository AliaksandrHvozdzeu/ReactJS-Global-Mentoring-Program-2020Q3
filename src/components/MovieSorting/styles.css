.sorting-title {
  color: #d6cbcb;
  font-weight: 100;
}

.sorting-select {
  background-color: var(--grey-darker);
  color: var(--white);
  font-size: 1.2rem;
  outline: none;
  margin-left: 2rem;
  text-transform: uppercase;
}

.sorting-select:hover {
  cursor: pointer;
}

.sorting-select::after {
  content: '';
  border-top: 5px solid var(--red);
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-bottom: 5px solid transparent;
  position: absolute;
  top:50%;
  left: 50%;
  z-index: 2;
}
