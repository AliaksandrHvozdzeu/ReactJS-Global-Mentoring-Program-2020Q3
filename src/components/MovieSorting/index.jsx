import React, { useState } from 'react';
import PropTypes from 'prop-types';
import './styles.css';

const MovieSorting = ({title, options, onAction, selectedOption}) => {
  const [selected, setSelected] = useState(selectedOption || options[0]);
  const onChange = (event) => {
    const { value } = event.target;
    onAction(value);
    setSelected(value);
  };
  return (
    <label htmlFor="date-filter" className="sorting-title">
      {title}
      <select id="date-filter" className="sorting-select" onChange={onChange} value={selected}>
        {options.map((option) => (
          <option key={option} className="sorting-item" value={option}>
            {option}
          </option>
        ))}
      </select>
    </label>
  );
};

MovieSorting.propTypes = {
  title: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(PropTypes.string).isRequired,
  onAction: PropTypes.func.isRequired,
  selectedOption: PropTypes.string,
};

MovieSorting.defaultProps = {
  selectedOption: null
};

export default MovieSorting;
