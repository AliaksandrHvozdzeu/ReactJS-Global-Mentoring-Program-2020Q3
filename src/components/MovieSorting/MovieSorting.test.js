import React from "react";
import rerender from "react-test-renderer";
import MovieSorting from "./js/MovieSorting";
import { jest } from "@jest/globals";
import { shallow } from "enzyme";

describe("when MovieSorting", () => {

  const title = "title";
  const options = ['All', 'Documentary', 'Comedy', 'Horror', 'Crime'];
  const onAction = jest.fn();
  const selectedOption = "";

  test("then snapshot created", () => {
    const component = rerender.create(<MovieSorting title={title}
                                                    options={options}
                                                    onAction={onAction}
                                                    selectedOption={selectedOption} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('onChange', () => {
    const component = shallow(<MovieSorting title={title}
                                            options={options}
                                            onAction={onAction}
                                            selectedOption={selectedOption} />);
    const selector = component.find('#date-filter');
    const event = {
      preventDefault() {},
      target: { value: 'the-value' }
    };
    selector.simulate('change', event.target);
    const elem = component.find('.sorting-item');
    expect(elem.length).toBe(5);
  });

});
