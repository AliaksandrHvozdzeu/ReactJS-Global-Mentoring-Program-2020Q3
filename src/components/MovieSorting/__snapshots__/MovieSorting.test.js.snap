// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when MovieSorting then snapshot created 1`] = `
<label
  className="sorting-title"
  htmlFor="date-filter"
>
  title
  <select
    className="sorting-select"
    id="date-filter"
    onChange={[Function]}
    value="All"
  >
    <option
      className="sorting-item"
      value="All"
    >
      All
    </option>
    <option
      className="sorting-item"
      value="Documentary"
    >
      Documentary
    </option>
    <option
      className="sorting-item"
      value="Comedy"
    >
      Comedy
    </option>
    <option
      className="sorting-item"
      value="Horror"
    >
      Horror
    </option>
    <option
      className="sorting-item"
      value="Crime"
    >
      Crime
    </option>
  </select>
</label>
`;
